metagraph/src/cli/align.cpp

```diff
@@ -426,21 +414,29 @@ int align_to_graph(Config *config) {
             }
 
             auto process_batch = [&,graph](SeqBatch batch) {
-                // make a shared_ptr in a thread-safe way
-                std::shared_ptr<DeBruijnGraph> aln_graph(
-                    std::shared_ptr<DeBruijnGraph>{}, graph.get()
-                );
+                // make an aliased shared_ptr in a thread-safe way
+                auto aln_graph = std::shared_ptr<DeBruijnGraph>(std::shared_ptr<DeBruijnGraph>{}, graph.get());
+
+                if (aln_graph->get_mode() == DeBruijnGraph::PRIMARY)
+                    aln_graph = std::make_shared<CanonicalDBG>(aln_graph, 100'000);
```

Comment:
Isn't this already wrapped in line 337?

Reply:
yes, that's the case. In a different branch, I have an if-statement before this which checks if the graph can be cached, then wraps the base graph if that's the case. So, by the time it gets here, you could be left with a primary graph, so this is meant to fix that case.

This statement is left over from that, so it should have been spliced out.